scripts/dev/celeritas-gen.py

```diff
@@ -501,16 +501,26 @@ class {name}Test : public ::celeritas::test::Test
 }
 
 
-def generate(root, filename, namespace):
+def removed_toplevel(filestr):
+    """Strip the leading directory from filestr.
+    """
+    return filestr.partition(os.path.sep)[-1]
+
+def generate(repodir, filename, namespace):
     if os.path.exists(filename):
         print("Skipping existing file " + filename)
         return
 
-    relpath = os.path.relpath(filename, start=root)
-    dirname = os.path.dirname(relpath)
+    dirname = os.path.relpath(filename, start=repodir)
```

Comment:
For the next statement to work properly, the path has to be somewhat 'normalize'.  Do we know if `relpath` return a normalized path or do we need
```suggestion
    filename = os.path.normpath(filename)  # In case the user 'typed' (intentionally or not)  somepath/../../../sources/celeritas/...
    dirname = os.path.relpath(filename, start=repodir)
```

Reply:
As long as `basedir` is a parent of `filename` (which is assumed since it should be the repository's root), this will work. Do we need to support the use case of generating a file in celeritas when (say) acceleritas is the current working directory? (This would require putting the "basename" calculation inside of the filename loop rather than outside, since users could do `celeritas-gen.py src/foo.cc ../celeritas/bar.cc` ... which is dumb)

Reply:
Note: 'relpath' does internally call `normpath`:
```
>>> os.path.relpath("abc/123/../234/xyz", "abc")
'234/xyz'
```
so there is actually nothing to do to accomplish what I was hope for (avoid spurrious indirection in the path being printed out).

>  Do we need to support ... This would require putting ...

I agree with you that this script does not need to become that complicated for that little gain (albeit, hopefully it does not fail too badly)

and it does fail 'fine':

```
-bash-4.2$ ../scripts/dev/celeritas-gen.py demo-loop/foo.cc ../celeritas/bar.cc
Traceback (most recent call last):
  File "../scripts/dev/celeritas-gen.py", line 596, in <module>
    main()
  File "../scripts/dev/celeritas-gen.py", line 592, in main
    generate(repodir, fn, args.namespace)
  File "../scripts/dev/celeritas-gen.py", line 564, in generate
    with open(filename, 'w') as f:
FileNotFoundError: [Errno 2] No such file or directory: '../celeritas/bar.cc'
```

so all good.

Reply:
except that the example was not quite 'right':
```
-bash-4.2$ ../scripts/dev/celeritas-gen.py demo-loop/foo.cc ../src/celeritas/bar.cc
-bash-4.2$ ls -l demo-loop/foo.cc ../src/celeritas/bar.cc
-rw-r--r-- 1 pcanal nobody 659 Nov 29 11:50 demo-loop/foo.cc
-rw-r--r-- 1 pcanal nobody 659 Nov 29 11:50 ../src/celeritas/bar.cc
-bash-4.2$ grep file ../src/celeritas/bar.cc
// See the top-level COPYRIGHT file for details.
//! \file celeritas/bar.cc
```
so actually it seems to 'already' work good enough (not that it matters :) )

Reply:
Yeah, it works now from inside the same "celeritas" directory (whereas previously it did not), which is what I intended to fix with this :)

Reply:
FYI in the last 2 attempts I was in the 'app' folder ;)

Reply:
I meant from *anywhere* inside the celeritas directory ;)